From 189edf3b5dd5c48a0eeeafcf2c60504a6f4ceac7 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Tue, 14 Dec 2021 11:49:32 +0200
Subject: [PATCH] Revert "[SW-63186] gaudi2: security: set axprot for edma"

Moving axprot configuration of edma to zephyr post isolation.
This is required as the engines which have isolation logic must
be accessed/configured post isolation configuration.

This reverts commit 8ecd7c75b5a61498f0ff96110731e7671cc922da.

Change-Id: I12cd559656953d217fecbae3dc541d59ad96e0ee
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 995c7b2..c839c75 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -48,8 +48,6 @@
 #include <dcore0_mme_wb0_mstr_if_core_hbw_regs.h>
 #include <dcore0_mme_ctrl_mstr_if_core_hbw_regs.h>
 #include <dcore0_mme_ctrl_mstr_if_core_lbw_regs.h>
-#include <dcore0_edma0_mstr_if_core_hbw_regs.h>
-#include <dcore0_edma0_mstr_if_core_lbw_regs.h>
 #include <pdma0_mstr_if_core_hbw_regs.h>
 #include <pdma0_mstr_if_core_lbw_regs.h>
 
@@ -216,7 +214,6 @@
 	AXPROT_HMMU,
 	AXPROT_TPC,
 	AXPROT_MME,
-	AXPROT_EDMA,
 	AXPROT_PDMA,
 };
 
@@ -948,10 +945,6 @@
 #define MME_INST_NUM		NUM_OF_DCORES
 #define MME_INST_DIST		DCORE_DIST
 #define MME_CRTL_RTR_INST_NUM	1
-#define EDMA_RTR_INST_NUM	2
-#define EDMA_RTR_INST_DIST	0x10000
-#define EDMA_INST_NUM		NUM_OF_DCORES
-#define EDMA_INST_DIST		DCORE_DIST
 #define PDMA_RTR_INST_NUM	1
 #define PDMA_RTR_INST_DIST	0
 #define PDMA_INST_NUM		2
@@ -997,14 +990,6 @@
 		.dcore_num = MME_INST_NUM,
 		.inst_dist = MME_SBTE_RTR_INST_DIST,
 		.inst_num = MME_SBTE_RTR_INST_NUM,
-	},
-	[AXPROT_EDMA] = {
-		.base_reg = mmDCORE0_EDMA0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
-		.value = AXPROT_USER_VAL,
-		.dcore_dist = EDMA_INST_DIST,
-		.dcore_num = EDMA_INST_NUM,
-		.inst_dist = EDMA_RTR_INST_DIST,
-		.inst_num = EDMA_RTR_INST_NUM,
 	},
 	[AXPROT_PDMA] = {
 		.base_reg = mmPDMA0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
@@ -1127,18 +1112,6 @@
 	preboot_axprot_regs_init(AXPROT_MME);
 }
 
-static void preboot_secure_edma(void)
-{
-	/* Configure edma to user level */
-	/* HBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_EDMA);
-
-	axprot_reg[AXPROT_EDMA].base_reg = mmDCORE0_EDMA0_MSTR_IF_CORE_LBW_AXPROT_OVRD;
-
-	/* LBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_EDMA);
-}
-
 static void preboot_secure_pdma(void)
 {
 
@@ -1168,7 +1141,6 @@
 		preboot_secure_trace();
 		preboot_secure_tpc();
 		preboot_secure_mme();
-		preboot_secure_edma();
 		preboot_secure_pdma();
 		preboot_secure_lbw();
 		preboot_secure_hbw();
